vortex-array/src/data/mod.rs

```diff
@@ -132,6 +132,50 @@ impl Array {
         // for constructing an Array, e.g. `try_new_owned`.
         array.vtable().validate(&array)?;
 
+        // Validate that the ArrayVisitor correctly returns the number of buffers and children
+        #[cfg(debug_assertions)]
```

Comment:
In general there should be something that asserts well-formedness of an implementation. This one is annoying since we must produce an array of given kind in order to validate it.